File: AmadeusBack/amadeus-api/database/models/Project.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace amadeus_api.database.models;

public class Project : Entity
{
    public string? Description { get; set; }
    public string? Summary { get; set; }
    public string? Code { get; set; }
    public ProjectStatus Status { get; set; } = ProjectStatus.NEW;

    [ForeignKey(nameof(Owner))]
    public long OwnerId { get; set; }
    public User Owner { get; set; } = null!;

    [ForeignKey(nameof(Customer))]
    public long? CustomerId { get; set; }
    public Customer? Customer { get; set; }

    [ForeignKey(nameof(CMDB))]
    public long? CMDBId { get; set; }
    public CMDB? CMDB { get; set; }

    [ForeignKey(nameof(Category))]
    public long? CategoryId { get; set; }
    public ProjectCategory? Category { get; set; }

    public List<ProjectPhase>? Phases { get; set; }
    public List<ProjectTask>? Tasks { get; set; }



    override public void MarkDeleted(long? userId)
    {
        base.MarkDeleted(userId);
        Status = ProjectStatus.DELETED;
    }
}

File: AmadeusFront/src/app/models/Project.ts
import { Customer } from "./Customer";
import { Entity } from "./Entity";
import { User } from "./User";

export interface Project extends Entity {
    code: string | null;
    description?: string;
    summary?: string;
    status: ProjectStatus;
    owner: User | null;
    customer: Customer | null;
}

export enum ProjectStatus {
    NEW = 'NEW',
    IN_PROGRESS = 'IN_PROGRESS',
    COMPLETED = 'COMPLETED',
    CANCELLED = 'CANCELLED',
    DELETED = 'DELETED',
    ARCHIVED = 'ARCHIVED'
}